addons/block-switching/userscript.js

```diff
@@ -669,111 +669,41 @@ export default async function ({ addon, global, console, msg }) {
     }, 0);
   };
 
-  const customContextMenuHandler = function (options) {
-    if (addon.self.disabled) return;
-
-    if (addon.settings.get("border")) {
-      addBorderToContextMenuItem = options.length;
-    }
-
-    if (this._originalCustomContextMenu) {
-      this._originalCustomContextMenu.call(this, options);
-    }
-
-    const switches = blockSwitches[this.type];
-    for (const opcodeData of switches) {
-      const isNoop = opcodeData.opcode === "noop";
-      if (isNoop && !addon.settings.get("noop")) {
-        continue;
-      }
-      const translationOpcode = isNoop ? this.type : opcodeData.opcode;
-      const translation = msg(translationOpcode);
-      options.push({
-        enabled: true,
-        text: translation,
-        callback: menuCallbackFactory(this, opcodeData),
-      });
-    }
-  };
-
-  const injectCustomContextMenu = (block) => {
-    const type = block.type;
-    if (!Object.prototype.hasOwnProperty.call(blockSwitches, type)) {
-      return;
-    }
-
-    if (block._customContextMenuInjected) {
-      return;
-    }
-    block._customContextMenuInjected = true;
-
-    if (block.customContextMenu) {
-      block._originalCustomContextMenu = block.customContextMenu;
-    }
-
-    block.customContextMenu = customContextMenuHandler;
-  };
-
-  const changeListener = (change) => {
-    if (change.type !== "create") {
-      return;
-    }
-
-    for (const id of change.ids) {
-      const block = Blockly.getMainWorkspace().getBlockById(id);
-      if (!block) continue;
-      injectCustomContextMenu(block);
-    }
-  };
-
-  const mutationObserverCallback = (mutations) => {
-    if (addon.self.disabled) return;
-    for (const mutation of mutations) {
-      for (const node of mutation.addedNodes) {
-        if (node.classList.contains("blocklyContextMenu")) {
-          if (addBorderToContextMenuItem === -1) {
-            continue;
-          }
-          const children = node.children;
-          const item = children[addBorderToContextMenuItem];
-          if (item) {
-            item.style.paddingTop = "2px";
-            item.style.borderTop = "1px solid hsla(0, 0%, 0%, 0.15)";
-          }
-          addBorderToContextMenuItem = -1;
+  const Blockly = await addon.tab.traps.getBlockly();
```

Comment:
Wouldn't it make more sense to assign Blockly to the `await addon.tab.traps.getBlockly()` on line 4?

Reply:
huh, yeah, didnt see that